crates/vm/src/arch/segment.rs

```diff
@@ -157,12 +79,42 @@ where
     pub metrics: VmMetrics,
 }
 
-pub struct ExecutionSegmentState {
-    pub pc: u32,
-    pub is_terminated: bool,
+pub struct TracegenCtx {
+    pub timestamp: u32,
```

Comment:
fyi: timestamp seems like it will need to be inside `TracingMemory` later since that is what really manages it most of time, but I expect we move `Streams` inside `Ctx` eventually.

Reply:
okay. i was wondering if it was possible to keep all execution state here and make all the controllers etc. stateless, and take a reference to this vm state as argument